### DISCUSSION TITLE
Packets not arriving

### DISCUSSION THREAD

**Discussion by orangepineapple** [ANLAB-KAIST/KENSv3#201] [Project Q&A 2. Transport Layer (TCP)]
I have been trying to figure out why packetArrive is not being called, and I am a bit stuck. Any hints would be appreciated. Thank you!
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/57548877/421d6a70-0eaa-4eb2-a6d0-2e0085d8f402)

**Comment by Pusnow** [comment]
First of all, your packet's length is zero. You should fill proper value for the length field.

I suggest you to try to send a SYN packet, which contains correct IP addresses, TCP ports, and other metadata including the length field and checksum.


And also, your `packetArrived` method is not always called, because we test your TCP implementation with TA's implementation. Please, refer to previous discussions such as #199 .
